internal/expr/parser.go

```
@@ -265,28 +272,174 @@ func (p *Parser) parseIdentifier() (string, bool) {
 	return "", false
 }
 
+// parseColumn parses a column made up of name bytes, optionally dot-prefixed by
+// its table name. parseColumn returns an error so that it can be used with
```

The code doesn't seem to return an error :) Perhaps we need to replace `nil` with an `error` in error cases.

---

Ah this is saying it returns an error type, i.e. the parseList function takes this function as an argument so the type that parseList wants must match the type of this function. Even though we never actually need to return an error from this function.